File: Benchmarking/src/gen_106_19.swift
  import Dispatch
  @inline(never)
  func f0(_ v0: [[Double]], _ v1: [Double], _ v2: Double) -> Double {
    let v9: [Double] = v0[1]
    var v4: [Double] = v1
    let v3: Double = v9[0]
    let v11: Double = v2 * v3
    let v8: [Double] = v0[0]
    var v13: Double = v2
    let v18: Double = v3 - v11
    var v20: [Double] = v1
    v4[0] = v3
    v20[2] = v11
    var v23: Double = v18
    let v45: Double = v20[0]
    let v24: Double = v1[2]
    v20[0] = v24
    let v46: Double = v8[0]
    let v34: [[Double]] = [v1, v20, v20, v20, v4]
    var v72: [[Double]] = v34
    var v43: Double = v23
    let v44: Double = v13 - v45
    v72[3] = v1
    v20[2] = v13
    v20[0] = v46
    let v57: [Double] = v72[0]
    var v48: Double = v11
    var v81: Double = v48
    v4[2] = v11
    v20[0] = v81
    v4[1] = v44
    v20[1] = v43
    let v161: Double = v57[2]
    return v161
  }
  func benchmark() {
    let v0: [[Double]] = [[0.0], [1.0], [2.0]]
    let v1: [Double] = [3.0, 4.0, 5.0]
    let v2: Double = 6.0
    let start = DispatchTime.now().uptimeNanoseconds
    var result: Double = 6.0
    for _ in 1...1000 {
      result = f0(v0, v1, v2)
    }
    let end = DispatchTime.now().uptimeNanoseconds
    print(result)
    print(end - start)
  }
  benchmark()

File: Benchmarking/src/gen_106_19.scala
import java.lang.System.nanoTime
import scala.collection.immutable.Vector
object Gen extends App {
  @noinline
  def f0(v0: Vector[Vector[Double]], v1: Vector[Double], v2: Double): Double = {
    val v9: Vector[Double] = v0(1)
    var v4: Vector[Double] = v1
    val v3: Double = v9(0)
    val v11: Double = v2 * v3
    val v8: Vector[Double] = v0(0)
    var v13: Double = v2
    val v18: Double = v3 - v11
    var v20: Vector[Double] = v1
    v4 = v4.updated(0, v3)
    v20 = v20.updated(2, v11)
    var v23: Double = v18
    val v45: Double = v20(0)
    val v24: Double = v1(2)
    v20 = v20.updated(0, v24)
    val v46: Double = v8(0)
    val v34: Vector[Vector[Double]] = Vector(v1, v20, v20, v20, v4)
    var v72: Vector[Vector[Double]] = v34
    var v43: Double = v23
    val v44: Double = v13 - v45
    v72 = v72.updated(3, v1)
    v20 = v20.updated(2, v13)
    v20 = v20.updated(0, v46)
    val v57: Vector[Double] = v72(0)
    var v48: Double = v11
    var v81: Double = v48
    v4 = v4.updated(2, v11)
    v20 = v20.updated(0, v81)
    v4 = v4.updated(1, v44)
    v20 = v20.updated(1, v43)
    val v161: Double = v57(2)
    v161
  }
  def benchmark(): Unit = {
    val v0: Vector[Vector[Double]] = Vector(Vector(0.0), Vector(1.0), Vector(2.0))
    val v1: Vector[Double] = Vector(3.0, 4.0, 5.0)
    val v2: Double = 6.0
    val start = nanoTime()
    var result: Double = 6.0
    (1 to 1000).foreach { _ =>
      result = f0(v0, v1, v2)
    }
    val end = nanoTime()
    println(result)
    println(end - start)
  }
  benchmark()
}